cedar-policy-formatter/src/pprint/doc.rs

```diff
@@ -151,541 +151,604 @@ impl Doc for ASTNode<Option<Expr>> {
+                .0
+                .group(),
+        )
     }
 }
 
 impl Doc for MultOp {
-    fn to_doc(&self, _: &mut Context<'_>) -> RcDoc<'_> {
-        RcDoc::text(self.to_string())
+    fn to_doc(&self, _: &mut Context<'_>) -> Option<RcDoc<'_>> {
+        Some(RcDoc::text(self.to_string()))
     }
 }
 
 impl Doc for ASTNode<Option<Mult>> {
-    fn to_doc(&self, context: &mut Context<'_>) -> RcDoc<'_> {
-        let e = self.as_inner().unwrap();
+    fn to_doc(&self, context: &mut Context<'_>) -> Option<RcDoc<'_>> {
+        let e = self.as_inner()?;
         let initial = &e.initial;
         let extended = &e.extended;
         // Convert a list of arithmetic expressions to a doc separated by the operators
         // We keep the convention that operators are on the right but do not put a subexpression on a single line.
-        extended
-            .iter()
-            .fold((initial.to_doc(context), initial), |doc, (op, e)| {
-                (
-                    doc.0
-                        .append(RcDoc::space())
-                        .append(add_comment(
-                            op.to_doc(context),
-                            get_comment_after_end(doc.1.info.0.end, &mut context.tokens),
-                            RcDoc::nil(),
+        Some(
+            extended
+                .iter()
+                .fold(
+                    Some((initial.to_doc(context)?, initial)),
+                    |pair, (op, e)| {
+                        let pair = pair?;
+                        Some((
+                            pair.0
+                                .append(RcDoc::space())
+                                .append(add_comment(
+                                    op.to_doc(context)?,
+                                    get_comment_after_end(pair.1.info.0.end, &mut context.tokens)?,
+                                    RcDoc::nil(),
+                                ))
+                                .append(RcDoc::line())
+                                .append(e.to_doc(context)),
+                            e,
                         ))
-                        .append(RcDoc::line())
-                        .append(e.to_doc(context)),
-                    e,
-                )
-            })
-            .0
-            .group()
+                    },
+                )?
+                .0
+                .group(),
+        )
     }
 }
 
 impl Doc for ASTNode<Option<Unary>> {
-    fn to_doc(&self, context: &mut Context<'_>) -> RcDoc<'_> {
-        let e = self.as_inner().unwrap();
+    fn to_doc(&self, context: &mut Context<'_>) -> Option<RcDoc<'_>> {
+        let e = self.as_inner()?;
         if let Some(op) = e.op {
             match op {
+                // PANIC SAFETY: policies with too many unary consective unary operatos should be rejected by our parser.
+                #[allow(clippy::unreachable)]
                 NegOp::OverBang | NegOp::OverDash => unreachable!("invalid policy!"),
```

Comment:
This function already returns an `Option`. Any hard in making this return `None` instead of crashing?

Reply:
Good point. Let me fix them.